presto-main/src/main/java/com/facebook/presto/execution/SqlStageExecution.java

```diff
@@ -192,31 +191,32 @@ public StageInfo getStageInfo()
                 ImmutableList::of);
     }
 
-    public synchronized void addExchangeLocation(ExchangeLocation exchangeLocation)
+    public synchronized void addExchangeLocations(PlanFragmentId fragmentId, Set<URI> exchangeLocations, boolean noMoreExchangeLocations)
     {
-        requireNonNull(exchangeLocation, "exchangeLocation is null");
-        RemoteSourceNode remoteSource = exchangeSources.get(exchangeLocation.getPlanFragmentId());
-        checkArgument(remoteSource != null, "Unknown remote source %s. Known sources are %s", exchangeLocation.getPlanFragmentId(), exchangeSources.keySet());
+        requireNonNull(noMoreExchangeLocations, "noMoreExchangeLocations is null");
```

Comment:
Why do you have this check for a ``boolean``? I am guessing you wanted to add it for ``exchangeLocations``?